app/src/screen.c

```diff
@@ -134,6 +140,50 @@ create_texture(SDL_Renderer *renderer, struct size frame_size) {
                              frame_size.width, frame_size.height);
 }
 
+// This may be called more than once to work around SDL bugs
+bool
+screen_init_renderer_and_texture(struct screen *screen) {
+    if (screen->texture != NULL) {
+      SDL_DestroyTexture(screen->texture);
+      screen->texture = NULL;
+    }
+    if (screen->renderer != NULL) {
+      SDL_DestroyRenderer(screen->renderer);
+      screen->renderer = NULL;
+    }
+
+    screen->renderer = SDL_CreateRenderer(screen->window, -1,
+                                            SDL_RENDERER_ACCELERATED);
+    if (!screen->renderer) {
+        LOGC("Could not create renderer: %s", SDL_GetError());
+        screen_destroy(screen);
+        return false;
+    }
+
+    if (SDL_RenderSetLogicalSize(screen->renderer, screen->frame_size.width,
+                                 screen->frame_size.height)) {
+        LOGE("Could not set renderer logical size: %s", SDL_GetError());
+        screen_destroy(screen);
+        return false;
+    }
+
+#ifdef HIDPI_SUPPORT
+    int window_w, window_h, renderer_w, renderer_h;
+    SDL_GetWindowSize(screen->window, &window_w, &window_h);
+    SDL_GetRendererOutputSize(screen->renderer, &renderer_w, &renderer_h);
```

Comment:
This function returns a status (which needs to be checked). And in practice, for me, it never succeeds (it returns a non-zero value) :/
I think [`SDL_GL_GetDrawableSize`](https://wiki.libsdl.org/SDL_GL_GetDrawableSize) is better for this (moreover, it may not fail).

Reply:
Great, I'll test with the SDL_GL_GetDrawableSize call.
What do you think of the idea of calling either and storing the one that does not fail?  I'm not sure what the other call will do if there is no GL renderer, but I'll test it.

Reply:
> What do you think of the idea of calling either and storing the one that does not fail?
`SDL_GL_GetDrawableSize()` may not fail (it returns `void`).